### PR TITLE
Fix the formatting and naming in examples.

### LINKED ISSUE #1889: Improve storage samples.
Cleanup the samples based on the comments by @frankyn.

### PR DESCRIPTION
**Pull request by coryan**
Let me know if I miss any.

This fixes #1889.

### CODE REVIEW

File: google/cloud/storage/examples/storage_object_samples.cc [hunk 1]
```diff
@@ -960,18 +1038,27 @@ void RewriteObjectNonBlocking(google::cloud::storage::Client client, int& argc,
...
-          std::cout << "Rewrote " << progress->total_bytes_rewritten << "/"
-                    << progress->object_size << std::endl;
-        });
-    std::cout << "Rewrote object " << meta->name() << " in bucket "
-              << meta->bucket() << "\nFull Metadata: " << *meta << std::endl;
+
+    StatusOr<gcs::ObjectMetadata> object_metadata =
+        rewriter.ResultWithProgressCallback(
+            [](StatusOr<gcs::RewriteProgress> const& progress) {
+              if (!progress.ok()) {
```
Comment (remyabel) [inline google/cloud/storage/examples/storage_object_samples.cc]: Should this be `if (!progress)`?

File: google/cloud/storage/examples/storage_object_samples.cc [hunk 2]
```diff
@@ -1039,18 +1129,26 @@ void RewriteObjectResume(google::cloud::storage::Client client, int& argc,
...
```
Comment (remyabel) [inline google/cloud/storage/examples/storage_object_samples.cc]: Same as above.

File: google/cloud/storage/examples/storage_bucket_samples.cc [hunk 3]
```diff
@@ -55,20 +55,24 @@ void ListBuckets(google::cloud::storage::Client client, int& argc,
...
     if (!bucket_list) {
         std::cerr << "Error reading bucket list for default project"
                   << ", status=" << bucket_list.status() << std::endl;
         return;
     }
-    for (auto&& meta : *bucket_list) {
-      if (!meta.ok()) {
+
+    for (auto&& bucket_metadata : *bucket_list) {
+      if (!bucket_metadata.ok()) {
```
Comment (remyabel) [inline google/cloud/storage/examples/storage_bucket_samples.cc]: Also here.

Comment (frankyn) [inline google/cloud/storage/examples/storage_bucket_samples.cc]: Following what the other changes have, `.ok()` needs to be removed. 

https://github.com/googleapis/google-cloud-cpp/blob/695675ce6c0ee169786e26388f4852032ba23235/google/cloud/storage/examples/storage_bucket_samples.cc#L66

https://github.com/googleapis/google-cloud-cpp/blob/695675ce6c0ee169786e26388f4852032ba23235/google/cloud/storage/examples/storage_bucket_samples.cc#L95

https://github.com/googleapis/google-cloud-cpp/blob/695675ce6c0ee169786e26388f4852032ba23235/google/cloud/storage/examples/storage_bucket_samples.cc#L204

https://github.com/googleapis/google-cloud-cpp/blob/695675ce6c0ee169786e26388f4852032ba23235/google/cloud/storage/examples/storage_bucket_samples.cc#L544

https://github.com/googleapis/google-cloud-cpp/blob/695675ce6c0ee169786e26388f4852032ba23235/google/cloud/storage/examples/storage_bucket_samples.cc#L862

File: google/cloud/storage/examples/storage_bucket_samples.cc [hunk 4]
```diff
@@ -87,15 +91,17 @@ void ListBucketsForProject(google::cloud::storage::Client client, int& argc,
   namespace gcs = google::cloud::storage;
   [](gcs::Client client, std::string project_id) {
     int count = 0;
-    for (auto&& meta : client.ListBucketsForProject(project_id)) {
-      if (!meta.ok()) {
+    for (auto&& bucket_metadata : client.ListBucketsForProject(project_id)) {
+      if (!bucket_metadata.ok()) {
```
Comment (remyabel) [inline google/cloud/storage/examples/storage_bucket_samples.cc]: And here.

File: google/cloud/storage/examples/storage_notification_samples.cc [hunk 5]
```diff
@@ -176,7 +182,7 @@ int main(int argc, char* argv[]) try {
 
   // Build the list of commands and the usage string from that list.
   using CommandType =
-      std::function<void(google::cloud::storage::Client, int&, char* [])>;
+      std::function<void(google::cloud::storage::Client, int&, char*[])>;
```
Comment (frankyn) [inline google/cloud/storage/examples/storage_notification_samples.cc]: In https://github.com/googleapis/google-cloud-cpp/blob/695675ce6c0ee169786e26388f4852032ba23235/google/cloud/storage/examples/storage_notification_samples.cc#L164

Question, Is `StatusOr<void>` not capable of `if (!var)`?

File: google/cloud/storage/examples/storage_object_samples.cc [hunk 6]
```diff
@@ -57,11 +57,12 @@ void ListObjects(google::cloud::storage::Client client, int& argc,
   namespace gcs = google::cloud::storage;
   [](gcs::Client client, std::string bucket_name) {
     for (auto&& object_metadata : client.ListObjects(bucket_name)) {
-      if (!object_metadata.ok()) {
+      if (!object_metadata) {
```
Comment (frankyn) [inline google/cloud/storage/examples/storage_object_samples.cc]: I see a few `.ok()` still remaining.  

https://github.com/googleapis/google-cloud-cpp/blob/695675ce6c0ee169786e26388f4852032ba23235/google/cloud/storage/examples/storage_object_samples.cc#L510

https://github.com/googleapis/google-cloud-cpp/blob/695675ce6c0ee169786e26388f4852032ba23235/google/cloud/storage/examples/storage_object_samples.cc#L547

https://github.com/googleapis/google-cloud-cpp/blob/695675ce6c0ee169786e26388f4852032ba23235/google/cloud/storage/examples/storage_object_samples.cc#L1045

https://github.com/googleapis/google-cloud-cpp/blob/695675ce6c0ee169786e26388f4852032ba23235/google/cloud/storage/examples/storage_object_samples.cc#L1136

**Pull request merged**